internal/commands/config_registries_default_test.go

```diff
@@ -0,0 +1,148 @@
+		)
+
+		it.Before(func() {
+			var err error
+			tmpDir, err = ioutil.TempDir("", "pack-home-*")
+			assert.Nil(err)
+
+			configFile = filepath.Join(tmpDir, "config.toml")
+		})
+
+		it.After(func() {
+			_ = os.RemoveAll(tmpDir)
+		})
+
+		when("list", func() {
+			it("returns official if none is set", func() {
+				command := commands.ConfigRegistriesDefault(logger, config.Config{}, configFile)
+				command.SetArgs([]string{})
+				assert.Succeeds(command.Execute())
+
+				assert.Contains(outBuf.String(), "official")
+			})
+
+			it("returns the default registry if one is set", func() {
+				command := commands.ConfigRegistriesDefault(logger, config.Config{DefaultRegistryName: "some-registry"}, configFile)
+				command.SetArgs([]string{})
+				assert.Succeeds(command.Execute())
+
+				assert.Contains(outBuf.String(), "some-registry")
+			})
+		})
+
+		when("set default", func() {
+			it("should set the default registry", func() {
+				cfg := config.Config{
+					Registries: []config.Registry{
+						{
+							Name: "myregistry",
+							URL:  "https://github.com/buildpacks/registry-index",
+							Type: "github",
+						},
+					},
+				}
+				command := commands.ConfigRegistriesDefault(logger, cfg, configFile)
+				command.SetArgs([]string{"myregistry"})
+				assert.Succeeds(command.Execute())
+
+				cfg, err := config.Read(configFile)
+				assert.Nil(err)
+
+				assert.Equal(cfg.DefaultRegistryName, "myregistry")
+			})
+
+			it("should fail if no corresponding registry exists", func() {
+				command := commands.ConfigRegistriesDefault(logger, config.Config{}, configFile)
+				command.SetArgs([]string{"myregistry"})
+				assert.Error(command.Execute())
+
+				output := outBuf.String()
+				assert.Contains(output, "no registry with the name 'myregistry' exists")
+			})
+
+			it("returns clear error if fails to write", func() {
+				assert.Nil(ioutil.WriteFile(configFile, []byte("something"), 0001))
+				cfg := config.Config{
+					Registries: []config.Registry{
+						{
+							Name: "myregistry",
+							URL:  "https://github.com/buildpacks/registry-index",
+							Type: "github",
+						},
+					},
+				}
+				command := commands.ConfigRegistriesDefault(logger, cfg, configFile)
+				command.SetArgs([]string{"myregistry"})
+				assert.ErrorContains(command.Execute(), "writing config to")
+			})
+		})
+
+		when("--unset", func() {
+			it("should unset the default registry, if set", func() {
+				command := commands.ConfigRegistriesDefault(logger, config.Config{DefaultRegistryName: "some-registry"}, configFile)
+				command.SetArgs([]string{"--unset"})
+				assert.Nil(command.Execute())
+
+				cfg, err := config.Read(configFile)
+				assert.Nil(err)
+				assert.Equal(cfg.DefaultRegistryName, "")
+				assert.Contains(outBuf.String(), fmt.Sprintf("Successfully unset default registry %s", style.Symbol("some-registry")))
+			})
+
+			it("should return an error if official registry is being unset", func() {
```

Comment:
The current implementation assumes that users are unable to remove the official registry, and thus, can't unset it, without replacing it for another default registry.